app/bundles/PageBundle/Model/PageModel.php

```diff
@@ -1038,6 +996,77 @@ public function getPageList($limit = 10, \DateTime $dateFrom = null, \DateTime $
         return $q->execute()->fetchAll();
     }
 
+    /**
+     * Get all params (e.g. UTM tags) from a url.
+     */
+    protected function getQueryFromUrl(string $pageUrl): array
+    {
+        $query             = [];
+        $urlQuery          = parse_url($pageUrl, PHP_URL_QUERY);
+        parse_str($urlQuery, $urlQueryArray);
+
+        if (empty($urlQueryArray)) {
+            return $query;
+        }
+
+        foreach ($urlQueryArray as $key => $value) {
+            $key           = strtolower($key);
+            $query[$key]   = urldecode($value);
+        }
+
+        return $query;
+    }
+
+    /**
+     * Set UTM Tags based on the query of a page hit.
+     *
+     * @return void
+     */
+    protected function setUtmTags(Hit $hit, Lead $lead)
```

Comment:
I see no reason for protected. Private has the great benefit that we can make "BC" changes and won't break anything.
```suggestion
    private function setUtmTags(Hit $hit, Lead $lead): void
```

Reply:
In this way we can not change it in a subclass (inherited). But I see your point and I will change it.